rmb-sdk-go/peer/peer.go

```diff
@@ -147,13 +162,58 @@ func NewPeer(
 		reader:  reader,
 		writer:  writer,
 		handler: handler,
+		encoder: e,
```

Comment:
I think the Peer need to also may have function `Encoder() Encoder` that returns the encoder. This allows handler to then use the encode to `decode` payloads. if they match expected schema.

```go
func (p *Peer) Encoder() Encoder {
   return p.encoder
}
```